src/utils/options.py

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/python
+
+"""
+    options
+    ~~~~~~~
+
+    Parse all configuration options from the config.ini file for the current site.
+    Additionally, parse the hdw.dat and restrict.dat files of the current site for 
+    other configuration information.
+
+    :copyright: 2023 SuperDARN Canada
+    :author: Theodore Kolkman
+"""
+
+import os
+import json
+from dataclasses import dataclass
+
+
+@dataclass
+class Options:
+    def __post_init__(self):
+        self.parse_config()  # Parse info from config file
+        self.parse_hdw()  # Parse info from hdw file
+        self.parse_restrict()  # Parse info from restrict.dat file
+        self.verify_options()  # Check that all parsed values are valid
+
+    # ZMQ Identities
+    radctrl_to_exphan_identity: str = "RADCTRL_EXPHAN_IDEN"
+    radctrl_to_dsp_identity: str = "RADCTRL_DSP_IDEN"
+    radctrl_to_driver_identity: str = "RADCTRL_DRIVER_IDEN"
+    radctrl_to_brian_identity: str = "RADCTRL_BRIAN_IDEN"
+    radctrl_to_dw_identity: str = "RADCTRL_DW_IDEN"
+    driver_to_radctrl_identity: str = "DRIVER_RADCTRL_IDEN"
+    driver_to_dsp_identity: str = "DRIVER_DSP_IDEN"
+    driver_to_brian_identity: str = "DRIVER_BRIAN_IDEN"
+    driver_to_mainaffinity_identity: str = "DRIVER_MAINAFFINITY_IDEN"
+    driver_to_txaffinity_identity: str = "DRIVER_TXAFFINITY_IDEN"
+    driver_to_rxaffinity_identity: str = "DRIVER_RXAFFINITY_IDEN"
+    mainaffinity_to_driver_identity: str = "MAINAFFINITY_DRIVER_IDEN"
+    txaffinity_to_driver_identity: str = "TXAFFINITY_DRIVER_IDEN"
+    rxaffinity_to_driver_identity: str = "RXAFFINITY_DRIVER_IDEN"
+    exphan_to_radctrl_identity: str = "EXPHAN_RADCTRL_IDEN"
+    exphan_to_dsp_identity: str = "EXPHAN_DSP_IDEN"
+    dsp_to_radctrl_identity: str = "DSP_RADCTRL_IDEN"
+    dsp_to_driver_identity: str = "DSP_DRIVER_IDEN"
+    dsp_to_exphan_identity: str = "DSP_EXPHAN_IDEN"
+    dsp_to_dw_identity: str = "DSP_DW_IDEN"
+    dspbegin_to_brian_identity: str = "DSPBEGIN_BRIAN_IDEN"
+    dspend_to_brian_identity: str = "DSPEND_BRIAN_IDEN"
+    dw_to_dsp_identity: str = "DW_DSP_IDEN"
+    dw_to_radctrl_identity: str = "DW_RADCTRL_IDEN"
+    dw_to_rt_identity: str = "RT_DW_IDEN"
+    rt_to_dw_identity: str = "DW_RT_IDEN"
+    brian_to_radctrl_identity: str = "BRIAN_RADCTRL_IDEN"
+    brian_to_driver_identity: str = "BRIAN_DRIVER_IDEN"
+    brian_to_dspbegin_identity: str = "BRIAN_DSPBEGIN_IDEN"
+    brian_to_dspend_identity: str = "BRIAN_DSPEND_IDEN"
+
+    def parse_config(self):
+        # Read in config.ini file for current site
+        if not os.environ["BOREALISPATH"]:
+            raise ValueError("BOREALISPATH env variable not set")
```

Comment:
Can you use the new `logging` way of raising errors? It would be nice to be consistent across the codebase.

Reply:
Since `log_config.py` imports this class, I don't think I would be able to use the new logging without causing circular dependence.